subsys/bluetooth/controller/hci/hci.c

```diff
@@ -744,6 +751,239 @@ static void read_bd_addr(struct net_buf *buf, struct net_buf **evt)
 	ll_addr_get(0, &rp->bdaddr.val[0]);
 }
 
+#if defined(CONFIG_BT_CTLR_CODEC_AND_DELAY_INFO)
+uint8_t __weak hci_vendor_read_std_codecs(
+	const struct bt_hci_std_codec_info_v2 **codecs)
+{
+	ARG_UNUSED(codecs);
+
+	/* return number of supported codecs */
+	return 0;
+}
+
+uint8_t __weak hci_vendor_read_vs_codecs(
+	const struct bt_hci_vs_codec_info_v2 **codecs)
+{
+	ARG_UNUSED(codecs);
+
+	/* return number of supported codecs */
+	return 0;
+}
+
+static void read_codecs(struct net_buf *buf, struct net_buf **evt)
+{
+	struct bt_hci_rp_read_codecs *rp;
+	const struct bt_hci_std_codec_info_v2 *std_codecs;
+	const struct bt_hci_vs_codec_info_v2 *vs_codecs;
+	size_t std_codecs_bytes;
+	size_t vs_codecs_bytes;
+	uint8_t num_std_codecs;
+	uint8_t num_vs_codecs;
+	uint8_t *ptr;
+	uint8_t i;
+
+	/* read standard codec information */
+	num_std_codecs = hci_vendor_read_std_codecs(&std_codecs);
```

Comment:
This would be certainly possible, but has the following consequences:
* New functions to poll the number of codecs to be able to allocate the response packet
* Separate vendor-specific functions for v1 and v2 of the commands
* Endianness-conversions become responsibility of the vendor-specific functions
Do you think the simplification here justifies these consequences?

Reply:
I missed the reuse of v2 interface for both versions of the response. Looking at the definition of these functions in the test, existing implementation simplifies vendor's definition of these interfaces.